marshmallow_utils/links.py

```diff
@@ -38,75 +69,101 @@ class LinksStore:
                 'draft': URITemplate('/api/records{/pid_value}/draft'),
             }
         })
+
+    If the ``config`` is provided in the constructor, links will be resolved
+    immediately. If no config is provided, the links will be resolved upon a
+    call to ``resolve()`` method with the config.
+
+    :params host: A hostname or a callable returning a hostname.
+    :params config: A link configuration/
+    :params context: Extra template variables.
+    :params ignore_missing: If true, missing templates will be ignored.
     """
 
-    def __init__(self, host=None, config=None):
+    def __init__(self, host=None, config=None, context=None,
+                 ignore_missing=True):
         """Constructor."""
         self._host = host
         self.config = config
         self._links = defaultdict(list)
+        self._context = context or {}
+        self._ignore_missing = ignore_missing
 
     def add(self, namespace, links):
         """Adds a dictionary of links under a namespace."""
         self._links[namespace].append(links)
+        if self.config:
+            self.resolve()
 
     @property
     def host(self):
         """Get the hostname."""
         return self._host() if callable(self._host) else self._host
 
-    def resolve(self, context=None, config=None):
+    def resolve(self, config=None, context=None):
```

Comment:
I should be able to not need `resolve` (do the expansion directly with `expand_link`)